File: DSA/LeetCode/DFS/2331-Evaluate-Boolean-Binary-Tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def evaluateTree(self, root):
        """
        :type root: Optional[TreeNode]
        :rtype: bool
        """
        if root.val == 2:
            return self.evaluateTree(root.left) or self.evaluateTree(root.right)
        elif root.val == 3:
            return self.evaluateTree(root.left) and self.evaluateTree(root.right)
        else:
                return bool(root.val)
        

File: DSA/LeetCode/DFS/2331-Evaluate-Boolean-Binary-Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var evaluateTree = function (root) {

    //DFS

    if (root.val == 2)
        return evaluateTree(root.left) || evaluateTree(root.right)
    else if (root.val == 3)
        return evaluateTree(root.left) && evaluateTree(root.right)
    else
        return !!root.val



};